src/fs/metadata_log.cc

```diff
@@ -100,18 +100,55 @@ void metadata_log::write_update(inode_info::file& file, inode_data_vec new_data_
     } else {
         cut_out_data_range(file, new_data_vec.data_range);
     }
-
     file.data.emplace(new_data_vec.data_range.beg, std::move(new_data_vec));
 }
 
 void metadata_log::cut_out_data_range(inode_info::file& file, file_range range) {
-    file.cut_out_data_range(range, [](inode_data_vec data_vec) {
-        (void)data_vec; // TODO: for compaction: update used inode_data_vec
+    file.cut_out_data_range(range, [&](const inode_data_vec& former, const inode_data_vec& left_remains, const inode_data_vec& right_remains) {
+        std::visit(overloaded {
```

Comment:
Iis this suggestion about _compacted_log_size or about data_cluster_contents, because in the first case I don't see how the "_compacted_log_size -= ..." lines could benefit from any comments

Reply:
Let me help you
```suggestion
        // Remove former before adding remains
        std::visit(overloaded {
```